src/commands/dsyms/__tests__/upload.test.ts

```diff
@@ -318,4 +329,29 @@ describe('execute', () => {
     )
     expect(output[10]).toContain('Handled 3 dSYMs with success')
   })
+
+  test('Should succeed with API key and site from datadog.json file', async () => {
+    const {context, code} = await runCLI('src/commands/dsyms/__tests__/fixtures/', {
+      configPath: 'src/commands/dsyms/__tests__/fixtures/datadog-ci.json',
+    })
+    const outputString = context.stdout.toString()
+    const output = outputString.split(EOL)
+
+    expect(outputString).not.toContain('Error')
+    expect(code).toBe(0)
+    expect(output[1]).toContain('Starting upload with concurrency 20. ')
+    expect(output[2]).toContain('Will look for dSYMs in src/commands/dsyms/__tests__/fixtures/')
+    expect(output[3]).toContain('Will use temporary intermediate directory: ')
+    expect(output[4]).toContain('Will use temporary upload directory: ')
+    expect(output[5]).toContain(
+      'Uploading C8469F85-B060-3085-B69D-E46C645560EA.zip (DDTest, arch: armv7, UUID: C8469F85-B060-3085-B69D-E46C645560EA)'
+    )
+    expect(output[6]).toContain(
+      'Uploading 06EE3D68-D605-3E92-B92D-2F48C02A505E.zip (DDTest, arch: arm64, UUID: 06EE3D68-D605-3E92-B92D-2F48C02A505E)'
+    )
+    expect(output[7]).toContain(
+      'Uploading 3BC12422-63CC-30E8-B916-E5006CE3286C.zip (DDTest, arch: arm64, UUID: 3BC12422-63CC-30E8-B916-E5006CE3286C)'
+    )
+    expect(output[10]).toContain('Handled 3 dSYMs with success')
```

Comment:
You may want to use [Jest snapshots](https://jestjs.io/docs/snapshot-testing) to ease testing of the output of the command.
It is being used for the synthetics command, cf. https://github.com/DataDog/datadog-ci/blob/74a41e248d1c21985129eb07a0ad817ba035c874/src/commands/synthetics/__tests__/reporters/default.test.ts#L216

Reply:
Good idea, but I am not sure it is worth doing it here for 3 tests.
As we have some random values and formatting in the output, it would require a bit of work to mock the randomness providers and make the snapshot readable enough so that it's easy to understand what broke in them.

I think snapshots are great for many things, they make it a lot easier to test big values and update them.
But that's also their drawback, that it's really easy to update, so there is less involvement in the process of updating them.
When you rewrite a line, you have more chances of thinking "Wait a minute, I should double check that this is what I want, doesn't seem right", whereas with snapshots you can enter a list of unrelated commands, but as you're not performing the change in itself you can more easily skip a bug.

So for a command that is not getting a lot of modifications, I am ok with these checks :)

Reply:
Yes, all good for me too!